docs/core/diagnostics/diagnostic-libraries.md

```diff
@@ -0,0 +1,161 @@
+| [📦 Microsoft.Extensions.Diagnostics.Probes](https://www.nuget.org/packages/Microsoft.Extensions.Diagnostics.Probes) | Support for Kubernetes environmental probes. |
+
+## Exception summarization
+
+When you're trying to generate meaningful diagnostic messages for exceptions without leaking sensitive data it's challenging. The standard exception message often lacks critical details that accompany the exception, while invoking the <xref:System.Exception.ToString%2A?displayProperty=nameWithType> method yields an excess of state information.
+
+The <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.IExceptionSummarizer> interface offers methods for extracting crucial details from recognized exception types, that are represented by a single `string`.
+
+The <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.IExceptionSummarizer.Summarize%2A?displayProperty=nameWithType> method traverses registered summarizers until it identifies one capable of handling the specific exception type. In the event that no summarizer is capable of recognizing the exception type, fallback summary is provided instead.
+
+### Example exception summarization usage
+
+The following example demonstrates how to use the `IExceptionSummarizer` interface to retrieve a summary of an exception.
+
+:::code source="snippets/exception-summary/Program.cs":::
+
+The preceding code:
+
+- Instantiates a new <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> instance, chaining a call to the <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.ExceptionSummarizationExtensions.AddExceptionSummarizer%2A> extension method.
+  - The `AddExceptionSummarizer` extension method accepts a delegate that is used to configure the `ExceptionSummarizerBuilder` instance.
+  - The `builder` is used to add the HTTP provider, which handles exceptions of type:
+    - <xref:System.OperationCanceledException>
+    - <xref:System.Threading.Tasks.TaskCanceledException>
+    - <xref:System.Net.Sockets.SocketException>
+    - <xref:System.Net.WebException>
+- Builds a new `ServiceProvider` instance from the `ServiceCollection` instance.
+- Gets an instance of the `IExceptionSummarizer` interface from the `ServiceProvider` instance.
+- Iterates over a collection of exceptions, calling the `Summarize` method on each exception and displaying the result.
+
+> [!WARNING]
+> The primary focus in the design of all exception summarization implementations is to provide diagnostic convenience to trusted users, rather than prioritizing the protection of personally identifiable information (PII). The <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.ExceptionSummary.Description?displayProperty=nameWithType> doesn't contain sensitive information, but the <xref:Microsoft.Extensions.Diagnostics.ExceptionSummarization.ExceptionSummary.AdditionalDetails?displayProperty=nameWithType> might contain sensitive information depending on the implementation.
```

Comment:
I don't know whay "provide diagnostic convenience to trusted users" means.

The goal of summarization is twofold:

* To eliminate privacy-sensitive information from exception state such that some meaningful exception information can be added to logs.
* To reduce the cardinality associated with exception state such that exceptions can be reliably counted in metrics. This matters since metric dimensions have limited cardinality.

Reply:
This language was adopted from this response here, https://github.com/dotnet/extensions/issues/4403#issuecomment-1718402588, but happy to update it.